azurelinuxagent/ga/exthandlers.py

```diff
@@ -294,53 +293,43 @@ def __last_gs_unsupported(self):
                self.__gs_aggregate_status.code == GoalStateAggregateStatusCodes.GoalStateUnsupportedRequiredFeatures
 
     def run(self):
-        etag, activity_id, correlation_id, gs_creation_time = None, None, None, None
-
         try:
-            extensions_goal_state = self.protocol.get_goal_state().extensions_goal_state
+            gs = self.protocol.get_goal_state()
+            egs = gs.extensions_goal_state
 
-            # self.ext_handlers and etag need to be initialized first, since status reporting depends on them; also
+            # self.ext_handlers needs to be initialized before returning, since status reporting depends on it; also
             # we make a deep copy of the extensions, since changes are made to self.ext_handlers while processing the extensions
-            self.ext_handlers = copy.deepcopy(extensions_goal_state.extensions)
-            etag = self.protocol.client.get_goal_state().incarnation
+            self.ext_handlers = copy.deepcopy(egs.extensions)
```

Comment:
Would it make sense to move this copying into the `self.protocol.get_goal_state()` call? We're concerned about changing the goal state object whilst processing the `ext_handlers`, but we're still leaving a mutable reference to the goal state object itself above (`gs` and `egs`).

Reply:
Can we add a `# TODO` to capture this in the code?

Reply:
it's a big TODO: v3 :)